src/forwarder/internal/pkg/cmd/run.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2020 Cisco and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"io/ioutil"
+	"net"
+	"net/url"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	nested "github.com/antonfisher/nested-logrus-formatter"
+
+	"github.com/networkservicemesh/sdk-vppagent/pkg/networkservice/chains/xconnectns"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/flags"
+	"github.com/networkservicemesh/sdk/pkg/tools/grpcoptions"
+
+	"github.com/networkservicemesh/cmd-forwarder-vppagent/src/forwarder/internal/pkg/vppinit"
+)
+
+func init() {
+	cmd := runCmd
+	rootCmd.AddCommand(cmd)
+	Flags(cmd.Flags())
+	cobra.OnInitialize(flags.FromEnv(flags.EnvPrefix, flags.EnvReplacer,cmd.Flags()))
+}
+
+var runCmd = &cobra.Command{
+	Use:   "run",
+	Short: "Runs xconnect network service",
+	Long: `Runs xconnect network service.  Supported mechanisms:
+     - memif
+     - kernel
+     - vxlan`,
+	Run: func(cmd *cobra.Command, args []string) {
+		// Capture signals to cleanup before exiting - note: this *must* be the first thing in main
+		c := make(chan os.Signal, 1)
+		signal.Notify(c,
+			os.Interrupt,
+			// More Linux signals here
+			syscall.SIGHUP,
+			syscall.SIGTERM,
+			syscall.SIGQUIT)
+
+		starttime := time.Now()
+		// Setup logging
+		logrus.SetReportCaller(true)
+		// TODO - decide whether or not to continue to use nested formatter
+		// nested formatter does *not* escape \n, and so backtraces in logs are usable
+		// during dev, but its probably undesirable for general use
+		logrus.SetFormatter(&nested.Formatter{})
+
+		// Context to use for all things started in main
+		ctx, cancelFunc := context.WithCancel(context.Background())
+		defer cancelFunc()
+
+		// grpc.ClientConn for dialing the vppagent
+		vppagentCC, err := grpc.DialContext(ctx, VppAgentURL.String(), grpc.WithInsecure())
```

Comment:
Should we add grpc.WithBlock()?

Reply:
I've been trying to minimize startup latency through the system... thus not having WithBlock() here.  I may be overly aggressive on that in places though... so totally willing to discuss it.